layers/best_practices_utils.cpp

```diff
@@ -1370,7 +1370,7 @@ bool BestPractices::ValidateCreateComputePipelineArm(const VkComputePipelineCrea
                                       kThreadGroupDispatchCountAlignmentArm);
     }
 
-    auto descriptor_uses = module_state->CollectInterfaceByDescriptorSlot(entrypoint_optional);
+    auto interface_variables = module_state->GetInterfaceVariable(entrypoint);
```

Comment:
Current code doesn't check if `module_state->GetInterfaceVariable(` returns a nullptr.

Also IMO the type returned by `GetInterfaceVariable` should be specified.

```suggestion
    const std::vector<InterfaceVariable>* interface_variables = module_state->GetInterfaceVariable(entrypoint);
    if (interface_variables) {
        ... // check for kVUID_BestPractices_CreateComputePipelines_ComputeSpatialLocality
    }
```

Reply:
We should probably have this discussion in the next team sync. After using `auto` _liberally_ for the past 6+ years, I haven't found a benefit to being verbose with types in cases like this, but that is my subjective opinion. If we want to reduce or remove the use of `auto`, that's fine, but we'll need to come to a consensus, and then going forward we can make that official. For now, this lines up with current usage of `auto` elsewhere in the codebase, so I don't see a need for that to change here.

Reply:
I made a case for not using auto here because to me it might have made it more clear to check for `nullptr`. My only indication that this was a pointer was it being dereferenced in the for loop. In general I don't mind `auto` (assuming my intellisense works).

Reply:
Updated suggestion.

```suggestion
    auto interface_variables = module_state->GetInterfaceVariable(entrypoint);
    if (interface_variables) {
        ... // check for kVUID_BestPractices_CreateComputePipelines_ComputeSpatialLocality
    }
```

Reply:
That's a good point. We've been putting off the "auto discussion" for a long time, and I'd like to get it nailed down on what the team would like to see going forward.

In this particular instance I personally don't see the benefit to even adding `*` as the type could change to `std::shared_ptr`, `std::optional`, or something else with `operator bool` and it does not change how the code below is interpreted by a human. Again, though, this is my lazy and subjective point of view, and I assume I'm in the minority here :)

Reply:
applied change, was going to add my two cents about `auto` but rather defer that to a proper discussion time/place